src/analysis/retail/shaman/restoration/modules/tier/TWW1TierSet.tsx

```diff
@@ -0,0 +1,151 @@
+  static AFFECTED_SPELLS = [
+    SPELLS.HEALING_SURGE,
+    TALENTS_SHAMAN.CHAIN_HEAL_TALENT,
+    TALENTS_SHAMAN.HEALING_WAVE_TALENT,
+  ];
+  protected combatants!: Combatants;
+  has4pc: boolean;
+  tidalWaves2pcBonusHealing: number = 0;
+  tidalWaves2pcOverHealing: number = 0;
+  tidalWaves4pcSavedMana: number = 0;
+  tidalWavesBuffedCastNumber: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active =
+      this.selectedCombatant.has2PieceByTier(TIERS.TWW1) &&
+      this.selectedCombatant.hasTalent(TALENTS_SHAMAN.TIDAL_WAVES_TALENT);
+    this.has4pc = this.selectedCombatant.has4PieceByTier(TIERS.TWW1);
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TWW1TierSet.AFFECTED_SPELLS),
+      this.onCast,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(TWW1TierSet.AFFECTED_SPELLS),
+      this.onHeal,
+    );
+  }
+
+  onHeal(event: HealEvent) {
+    // Check if player has been under Tidal Waves for at least 100ms to prevent misattributions in case of a Riptide at the end of a cast
+    if (
+      !this.selectedCombatant.hasBuff(
+        SPELLS.TIDAL_WAVES_BUFF.id,
+        event.timestamp,
+        0,
+        TIDAL_WAVES_BUFF_MINIMAL_ACTIVE_TIME,
+      )
+    ) {
+      return;
+    }
+
+    // If the caster is under Tidal Waves, tally the amount attributed to the 2pc bonus
+    this.tidalWaves2pcBonusHealing += calculateEffectiveHealing(event, TWW1_TIER_2PC_BONUS);
+    this.tidalWaves2pcOverHealing += calculateOverhealing(event, TWW1_TIER_2PC_BONUS);
+    this.tidalWavesBuffedCastNumber += 1;
+  }
+
+  onCast(event: CastEvent) {
+    if (this.selectedCombatant.hasBuff(SPELLS.INNERVATE.id)) {
+      return;
+    }
+
+    if (!event.resourceCost) {
+      return;
+    }
+
+    // Check if player has been under Tidal Waves for at least 100ms to prevent misattributions in case of a Riptide at the end of a cast
+    if (
+      !this.selectedCombatant.hasBuff(
+        SPELLS.TIDAL_WAVES_BUFF.id,
+        event.timestamp,
+        0,
+        TIDAL_WAVES_BUFF_MINIMAL_ACTIVE_TIME,
+      )
+    ) {
+      return;
+    }
+
+    if (this.has4pc) {
+      // 4pc bonus reduces mana cost by 8%
+      const baseCost = event.resourceCost[RESOURCE_TYPES.MANA.id];
+      this.tidalWaves4pcSavedMana += baseCost * TWW1_TIER_4PC_MANACOST_REDUCTION;
+    }
+  }
+
+  statistic() {
+    return (
+      <Statistic
+        size="flexible"
+        position={STATISTIC_ORDER.OPTIONAL(0)}
+        category={STATISTIC_CATEGORY.ITEMS}
+        tooltip={
+          <>
+            {formatNumber(this.tidalWavesBuffedCastNumber)} casts buffed with{' '}
+            <SpellLink spell={SPELLS.TIDAL_WAVES_BUFF} /> were improved by the 2-piece bonus,
+            resulting in <strong>{formatNumber(this.tidalWaves2pcBonusHealing)}</strong> bonus
+            healing ({formatNumber(this.tidalWaves2pcOverHealing)} overhealing)
+          </>
+        }
```

Comment:
should include a note in the tooltip that the crit/haste bonus is not accounted for

Reply:
It is done, with an optimistic "not yet" for the moment. I have an idea about how I want to go about this, but I'm not 100% confident it is really quantifiable.